uobtheatre/bookings/mutations.py

```diff
@@ -653,9 +627,57 @@ def resolve_mutation(cls, _, info, booking_reference, tickets, performance_id):
         return UnCheckInBooking(booking=booking, performance=performance)
 
 
+class CreateBookingTransfer(AuthRequiredMixin, SafeMutation):
+    """Mutation to transfer a booking to a difference performance. This creates
+    an inprogress booking (transfer) which is linked to the orignal booking via
+    transfered_to.
+
+    Args:
+        booking_id (str): The gloabl id of the Booking to transfer.
+        performance_id (str): The gloabl id of the performance which the
+            Booking should be transfered to.
+
+    Returns:
+        booking (BookingNode): The Booking for the new performance.
+
+    Raises:
+        GQLException: If the Payment was unsucessful.
+    """
+
+    booking = graphene.Field(BookingNode)
+
+    class Arguments:
+        booking_id = IdInputField(required=True)
+        performance_id = IdInputField(required=True)
+
+    @classmethod
+    def resolve_mutation(  # pylint: disable=too-many-arguments, too-many-branches
+        cls,
+        _,
+        info,
+        booking_id,
+        performance_id,
+    ):
+        booking = Booking.objects.get(pk=booking_id)
+
+        # TODO Not boxoffice
+        if booking.user.id != info.context.user.id and not info.context.user.has_perm(
+            "productions.boxoffice", booking.performance.production
+        ):
+            raise AuthorizationException(
+                message="You do not have permission to access this booking.",
+            )
```

Comment:
This should use an ability.
Prehaps https://github.com/BristolSTA/uobtheatre-api/blob/35154eb132f084b076f0b0d0aec39429ac7e7ac3/uobtheatre/productions/abilities.py#L52

and https://github.com/BristolSTA/uobtheatre-api/blob/35154eb132f084b076f0b0d0aec39429ac7e7ac3/uobtheatre/bookings/abilities.py#L5

Why can't box office operators transfer bookings?